packages/consumption/test/modules/requests/RequestsIntegrationTest.ts

```diff
@@ -53,7 +53,7 @@ export class RequestsTestsContext {
     public outgoingRequestsController: OutgoingRequestsController;
     public currentIdentity: CoreAddress;
     public mockEventBus = new MockEventBus();
-    public relationshipToReturnFromGetActiveRelationshipToIdentity: Relationship | undefined;
+    public relationshipToReturnFromGetRelationshipToIdentity: Relationship | undefined;
```

Comment:
Depends on if the tests for the relationship status are integration- or e2e-tests. The test for creating an outgoing request on new relationship is e2e-style and I have created the other relationship tests as e2e-runtime tests.

Reply:
e2e tests are not enough. We definitely also need these tests here.

Reply:
As an explanation why: it is way easier to track where errors come from in these tests than in full-fledged end2end tests. End2end tests are nice to have, but these tests saved the whole requests handling multiple times.

Reply:
Why are requests usually created the standard way rather than using the test object factory? If I wanted to put the relationship individually into each test where it is required, it would be confusing why it sometimes isn't there.

However, my favorite approach would be to do the relationship setup in the beforeEach of each test suite - it is more realistic anyway.

Reply:
by calling the create controller method

Reply:
> by calling the create controller method

I think it's easier that way

> However, my favorite approach would be to do the relationship setup in the beforeEach of each test suite - it is more realistic anyway.

I think this is ok if every test shares what kind of relationship should be returned in a define declaration.

Reply:
What is a define declaration?
Also - would putting the current relationship termination tests into consumption / transport mean that the runtime tests aren't necessary anymore? There's no substantial logic there. I'm generally not clear about the overall testing approach.

Reply:
sorry, i meant describe.

no, transport tests doesn't remove the need for runtime tests.